File: src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("pyton_test")
        self.counter = 0
        self.get_logger().info("Hello World")
        # 2Hz 
        self.create_timer(0.5, self.timer_callback)
    
    def timer_callback(self):
        self.get_logger().info(f"Hello Timer World {self.counter}")
        self.counter += 1
                                   

def main(args = None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)

    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyFirstNode : public rclcpp::Node {
public:
  MyFirstNode() : Node("cpp_test"),counter_(0) {
    RCLCPP_INFO(this->get_logger(), "Hello, world!");
    timer_ = this->create_wall_timer(std::chrono::seconds(1), 
        std::bind(&MyFirstNode::timer_callback, this));
  }
private:
  rclcpp::TimerBase::SharedPtr timer_;
  void timer_callback() {
    counter_++;
    RCLCPP_INFO(this->get_logger(), "Hello, world %d!", counter_);
  }
  int counter_;

};

int main(int argc, char **argv) {
  // Initialize the ROS node
  rclcpp::init(argc, argv);
  auto node = std::make_shared<MyFirstNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}